src/libs/antares/benchmarking/info_collectors.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include <string>
+#include <mutex>
+
+#include <antares/study.h>
+#include "file_content.h"
+
+
+namespace Benchmarking
+{
+
+struct OptimizationInfo
+{
+    unsigned int nbVariables = 0;
+    unsigned int nbConstraints = 0;
+    unsigned int nbNonZeroCoeffs = 0;
+};
+
+/*
+    === Info collectors ===
+*/
+
+class StudyInfoCollector
+{
+public:
+    StudyInfoCollector(const Antares::Data::Study& study)
+        : study_(study)
+    {}
+    void toFileContent(FileContent& file_content);
+private:
+    // Methods
+    void areasCountToFileContent(FileContent& file_content);
+    void linksCountToFileContent(FileContent& file_content);
+    void performedYearsCountToFileContent(FileContent& file_content);
+    void enabledThermalClustersCountToFileContent(FileContent& file_content);
+    void enabledBindingConstraintsCountToFileContent(FileContent& file_content);
+    void unitCommitmentModeToFileContent(FileContent& file_content);
+    void maxNbYearsInParallelToFileContent(FileContent& file_content);
+    void solverVersionToFileContent(FileContent& file_content);
+
+    // Member data
+    const Antares::Data::Study& study_;
+
+    // ----------------------------------------------------------------------------------------
+    // TODO : add some more pieces of information about study : see following data members.
```

Comment:
Do we want to add more info ? Or is this an old TODO note ?

Reply:
We wanted more pieces of information : these are pointed by the unused data members below the comment.
Do we still want them ?

Reply:
No, you may remove these data members.

Reply:
Done